README.md

```diff
@@ -42,7 +42,8 @@ $provider = new Stevenmaguire\OAuth2\Client\Provider\Salesforce([
     'redirectUri'       => 'https://example.com/callback-url'
 ]);
 
-$existingAccessToken = getAccessTokenFromYourDataStore();
+// A getAccessTokenFromYourDataStore() example. 
+$existingAccessToken->getAccessToken('password', ['username'=>'USERNAME', 'password' => 'PASSWORD']); 
```

Comment:
I don't think this improves clarity of the documentation - I think it actually makes it worse. This example appears to suggest that:

1. `$existingAccessToken` appears out of thin air
2. `$existingAccessToken` exposes a `getAccessToken` method similar to the provider itself

I think there is room to improve the documentation but I don't think this is it. 

`getAccessTokenFromYourDataStore()` is meant to serve as a placeholder for some function your application implements that retrieves an AccessToken instance from your own application persisted storage - whatever it may be.

Reply:
I think I see the confusion. 

Your documentation assumes a token already exists but does not specify how to generate that token. Not having a part of the documentation that states HOW one gets a token is what's confusing. 

When one looks up HOW to get a token the current Salesforce OAuth API documentation for REST ( https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/quickstart_oauth.htm ) states

> Using [the Salesforce] REST API requires an access token to successfully send requests. You can **get an access token** using the username-password authorization flow.

That's not a NEW access token. That's AN access token. 

I can see that since I re-used the variable $existingAccessToken  it would lead to that confusion. I've updated the README.md to perhaps clarify how your assumption that a token already existed was confusing to me and my reusing of the variable name might have be confusing to others.